releasenotes/notes/add-wrap-instructions-32335ea783203be9.yaml

```
@@ -0,0 +1,110 @@
+    want to register a callback function to the global ``WRAP_ANGLE_REGISTRY`` registry that will tell the
+    transpiler and :class:`.WrapAngles` pass how to adjust gates for angle bounds. The callback function
+    will take a list of  arbitrary ``float`` values representing the gate angles from the circuit, as well
+    as the qubit indices in the circuit the gate was operating on and it will return a :class:`.DAGCircuit`
+    that represents an equivalent circuit for the gate with that angle but respecting the angle bounds and
+    other :class:`.Target` constraints. For example::
+
+        import math
+
+        from qiskit.dagcircuit import DAGCircuit
+        from qiskit.transpiler import target
+        from qiskit.transpiler.passes.utils.wrap_angles import WRAP_ANGLE_REGISTRY
+
+        target = Target(num_qubits=1)
+        target.add_instruction(RZGate(Parameter("theta")), angle_bounds=[(-math.pi, math.pi)])
+
+        def callback(angles: List[float], qubits: List[int]) -> DAGCircuit:
+            """Callback function to wrap RZ gate angles
+
+            Args:
+                angles: The list of floating point parameter values for the instance of RZGate in
+                    the circuit
+                qubits: The physical qubit indices that this gate is operating on
+
+            Returns:
+                The DAGCircuit of the equivalent circuit
+
+            angle = angles[0]
+            dag = DAGCircuit()
+            dag.add_qubits([Qubit()])
+            if angle > 0:
+                divisor = math.pi
+            else:
+                divisor = -math.pi
+            gate_counts = int(angles[0] // divisor)
+            rem = angles[0] % divisor
+            for _ in range(gate_counts):
+                dag.apply_operation_back(RZGate(math.pi), [dag.qubits[0]], check=True)
+            dag.apply_operation_back(rem, [dag.qubits[0]], check=True)
+
+        WRAP_ANGLE_REGISTRY.add_wrapper("rz", callback)
+
+    :meth:`.Target.has_angle_bounds` can be used to check whether there are
+    any angle bounds set in the target, and :meth:`.Target.gate_has_angle_bounds`.
+
+    If you want to apply the angle bounds from a target to any gates in a circuit the
+    :class:`.WrapAngles` transpiler pass should be used to do this.
+  - |
+    Added a new kwarg ``check_angle_bounds`` to the :meth:`.Target.instruction_supported` method.
+    When set to ``True`` (the default) the method will check if the specified ``parameters``
+    also conforms to any angle bounds that may exist for the instruction being queried.
+  - |
+    Added a new transpiler pass :class:`.WrapAngles` which is used to apply
+    angle constraints on gates in the :class:`.Target`. If a :class:`.Target` has defined
+    angle bounds this pass will analyze all the parameters for the gates in the circuit and
+    check that the target. For example, if a target contains a custom gate that has angle bounds
```

```suggestion
    check that the target conforms to the bounds. For example, if a target contains a custom gate that has angle bounds
```
I think this sentence was left incomplete.

---

Updated for clarity in: https://github.com/Qiskit/qiskit/pull/14406/commits/f84003a9af4b9b4e1c907477f7f7e41049d80339